extensions/ql-vscode/src/variant-analysis/view-autofixes.ts

```diff
@@ -0,0 +1,801 @@
+  }
+  await ensureDir(autofixOutputStoragePath);
+
+  return {
+    variantAnalysisIdStoragePath,
+    sourceRootsStoragePath,
+    autofixOutputStoragePath,
+  };
+}
+
+/**
+ * Processes the selected repositories for autofix generation.
+ */
+async function processSelectedRepositories(
+  selectedRepoNames: string[],
+  variantAnalysisIdStoragePath: string,
+  sourceRootsStoragePath: string,
+  autofixOutputStoragePath: string,
+  localAutofixPath: string,
+  credentials: Credentials,
+  logger: NotificationLogger,
+): Promise<string[]> {
+  const outputTextFiles: string[] = [];
+  await Promise.all(
+    selectedRepoNames.map(async (nwo) =>
+      withProgress(
+        async (progressForRepo: ProgressCallback) => {
+          // Get the sarif file.
+          progressForRepo(progressUpdate(1, 3, `Getting sarif`));
+          const repoStoragePath = join(variantAnalysisIdStoragePath, nwo);
+          const sarifFile = await getSarifFile(repoStoragePath, nwo);
+
+          // Read the contents of the variant analysis' `repo_task.json` file,
+          // and confirm that the `databaseCommitSha` and `resultCount` exist.
+          const repoTask: VariantAnalysisRepositoryTask =
+            await readRepoTask(repoStoragePath);
+          if (!repoTask.databaseCommitSha) {
+            throw new Error(`Missing database commit SHA for ${nwo}`);
+          }
+          if (!repoTask.resultCount) {
+            throw new Error(`Missing variant analysis result count for ${nwo}`);
+          }
+
+          // Download the source root.
+          // Using `0` as the progress step to force a dynamic vs static progress bar.
+          // Consider using `reportStreamProgress` as a future enhancement.
+          progressForRepo(progressUpdate(0, 3, `Downloading source root`));
+          const srcRootPath = await downloadPublicCommitSource(
+            nwo,
+            repoTask.databaseCommitSha,
+            sourceRootsStoragePath,
+            credentials,
+            logger,
+          );
+
+          // Run autofix.
+          progressForRepo(progressUpdate(2, 3, `Running autofix`));
+          await runAutofixForRepository(
+            nwo,
+            sarifFile,
+            srcRootPath,
+            localAutofixPath,
+            autofixOutputStoragePath,
+            repoTask.resultCount,
+            logger,
+            outputTextFiles,
+          );
+        },
+        {
+          title: `Processing ${nwo}`,
+          cancellable: false,
+        },
+      ),
+    ),
+  );
+
+  return outputTextFiles;
+}
+
+/**
+ * Gets the path to a SARIF file in a given `repoStoragePath`.
+ */
+async function getSarifFile(
+  repoStoragePath: string,
+  nwo: string,
+): Promise<string> {
+  // Get results directory path.
+  const repoResultsStoragePath = join(repoStoragePath, "results");
+  // Find sarif file.
+  const sarifFiles = await glob(`${repoResultsStoragePath}/**/*.sarif`);
```

Comment:
I believe this should always live in `results/results.sarif`. However, can we add a method to `VariantAnalysisManager`/`VariantAnalysisResultsManager` that returns this path since those should be responsible for the result storage?

Perhaps it could even make sense to first call `variantAnalysisResultsManager.isVariantAnalysisRepoDownloaded` to ensure the SARIF file is downloaded so you can give a better error message.

Reply:
Updated in https://github.com/github/vscode-codeql/pull/4065/commits/1d203a4bd20c7878355fccc7f54307e4f16b32da.